bindata/network/openshift-sdn/sdn.yaml

```diff
@@ -178,6 +178,39 @@ spec:
             command: ["test", "-f", "/etc/cni/net.d/80-openshift-network.conf"]
           initialDelaySeconds: 5
           periodSeconds: 5
+{{- if .EnableConnectivityChecks }}
```

Comment:
This is useful for all network providers, right? It should probably be its own daemonset. It can be in the cluster-network-operator namespace, though.

Reply:
PodNetworkConnectivityCheck resources test connection from pods managed by the operator to a target endpoint.
Then check-endpoints tool is run in a sidecar container and that checks the resources defined to its pod and updates the status (did it resolve ok, did it connect ok).
The PR right now has as placeholder checks from every ovnkube-node to every kube-apiserver endpoint/svc and internal/external LBs which I tried to discuss on our last team meeting 3 days ago but we didn't have time.
I guess there is value to have these generic tests to key targets like kube/openshift apiserver endpoints/services and LBs but run it within the operator namespace in a pod running a dummy container.
But I also I guess there is value in keeping at least pod checks for every ovnkube-node to ovn DB to complement the dbchecker that checks the DB is healthy and stale raft members.

Thoughts?

@squeed @trozet @danwinship @dcbw @abhat

Reply:
Well, we can create multiple `PodNetworkConnectivityCheck` objects, right? Then we should certainly create ones for generic "the cluster network is working" tests.

If you think it's useful for us to probe that ovn-nbdb is serving, then we can create a second resource. However, given that it's host-network, it's probably sufficient to just set a ReadinessProbe...?

Reply:
For more context:

https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/stability-point-to-point-network-check.md

Reply:
Yeah, as a matter of fact the PR does create multiple PodNetworkConnectivityChecks object.
The question now is:

From which pod (a generic one non-hostnetwork from CNO namespace) to where.
kubeapiserver endpoints/svc AND openshift-apiserver ?
internal/external LBs are good also?
Anything else?

Reply:
Be careful, if you check external LB, we may fall afoul of offline / proxied clusters.